community/detectors/joomla_cve_2023_23752/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202323752/Cve202323752VulnDetector.java

```diff
@@ -0,0 +1,473 @@
+    matchedServices.stream()
+        .filter(NetworkServiceUtils::isWebService)
+        .forEach(
+            networkService -> {
+              ScanResults results = isServiceVulnerable(networkService);
+              if (results.isSuccessful()) {
+                detectionReport.addDetectionReports(
+                    buildDetectionReport(targetInfo, networkService, results));
+              }
+            });
+    return detectionReport.build();
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService, ScanResults results) {
+    StringBuilder ScanResultReport = new StringBuilder();
+
+    ScanResultReport.append("Full Leaked Response:\n")
+        .append(results.leakedResponse())
+        .append("\n");
+    ScanResultReport.append("The leaked credentials are: \n")
+        .append("Database Password:\n")
+        .append(results.dataBasePassword())
+        .append("\n")
+        .append("Database UserName:\n")
+        .append(results.dataBaseUsername())
+        .append("\n");
+
+    if (results.isPublicDatabaseHost()) {
+      ScanResultReport.append(
+              "The dataBase host is Accessible to Public Because it has a public IP address, "
+                  + "Attackers can leverage leaked DataBase credentials to login into your DataBase, The DataBase HostName is: ")
+          .append(results.dataBaseHost())
+          .append("\n");
+    }
+
+    if (results.compromisedAdminAccount()) {
+      ScanResultReport.append(
+              "Scanner has checked the credentials against Administrator login page "
+                  + "and Leaked credentials had used as a Joomla Administrator credentials")
+          .append("\n");
+    }
+
+    if (results.compromisedUserAccount()) {
+      ScanResultReport.append(
+              "Scanner has checked the credentials against Users login page "
+                  + "and Leaked credentials had used as a Joomla User credentials")
+          .append("\n");
+    }
+
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE_2023_23752"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("Joomla unauthorized access to webservice endpoints")
+                .setDescription(
+                    "CVE-2023-23752: An improper access check allows unauthorized access to"
+                        + " webservice endpoints. attacker can get critical information of database and webserver like passwords by this vulnerability")
```

Comment:
```suggestion
                .setTitle("Joomla unauthorized access to webservice endpoints")
                .setDescription(
                    "CVE-2023-23752: An improper access check allows unauthorized access to"
                        + " webservice endpoints. attacker can get critical information of database and webserver like passwords by this vulnerability")
```
I think this description is a bit generic. From the web I found that DB credentials could be exposed by this finding, but are there any other credentials that could be exposed too? If not, let's just explicitly say that DB creds could be exposed.